core/src/main/java/org/hibernate/ogm/compensation/impl/InvocationCollectingGridDialect.java

```diff
@@ -97,24 +98,14 @@ public void executeBatch(OperationsQueue queue) {
 			while ( operation != null ) {
```

Comment:
Ok, but the batch delegator is always running, right? What I mean is, _something_ that collects grid dialect invocations in an object-style approach. Ideally, we should have this only once, maybe implementing different public interfaces geared towards different usages, but only paying once the penalty for object allocation.